lib/puppet/parameter.rb

```diff
@@ -543,7 +543,7 @@ def to_s
   def self.format_value_for_display(value)
     if value.is_a? Array
       formatted_values = value.collect {|v| format_value_for_display(v)}.join(', ')
-      "[#{formatted_values}]"
+      "[#{formatted_values}]".encode(Encoding.default_external)
```

Comment:
Changing parameter formatting here seems wrong... need to better understand the strings being passed through here during `puppet resource user` as some are `UTF-8`, while others are `CP850` and this code crashes with encoding issues.

This is probably the wrong spot to address this, and there's likely an issue with how strings propagate from the ADSI user provider.

But this hammer works at the moment - more investigation needed!